src/pages/TransactionDetails/components/TransactionInfo/TransactionLogsPanel/TransactionLogsPanel.tsx

```diff
@@ -0,0 +1,64 @@
+import { DetailItem } from 'components';
+import {
+  AddressDetailItem,
+  EventsList,
+  ScrDetailItem
+} from 'pages/TransactionDetails/components';
+import { TransactionType } from 'types';
+
+export const TransactionLogsPanel = ({
+  transaction
+}: {
+  transaction: TransactionType;
+}) => {
+  return (
+    <>
+      {transaction.logs && (
+        <>
+          {transaction.logs.address !== undefined && (
+            <AddressDetailItem address={transaction.logs.address} />
+          )}
+          {transaction.logs.events && transaction.logs.events.length > 0 && (
+            <DetailItem title={<div className='item-title'>Events</div>}>
+              <EventsList
+                events={transaction.logs.events}
+                txHash={transaction.txHash}
+                id={transaction.logs?.id ?? 'events'}
+              />
+            </DetailItem>
+          )}
+        </>
+      )}
+      {transaction.results && transaction.results.length > 0 && (
+        <div className='row'>
+          {transaction.results.map((result, resultIndex) => {
```

Comment:
maybe this could be processed outside render...I'd apply the filtering outside the view and here just iterate through valid data

Reply:
`const txs = transaction.results.filter(tx => tx.log)`

Reply:
updated